src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -93,47 +121,86 @@ void Bank::evaluate_and_allocate() {
         }
         storage_guard.unlock();
 
-        ov::parallel_for(vec.size(), [&](std::size_t idx) {
-            const auto& lt = vec[idx];
+        std::vector<AllocatedTensor> uids_to_allocated(uid_count);
+
+        // No allocation needed
+        if (device_for_alloc == "CPU") {
+            ov::parallel_for(vec.size(), [&](std::size_t idx) {
+                const auto& lt = vec[idx];
+                std::unique_lock dev_guard(device_bank.mutex);
+                auto iter_device_registered = device_bank.registered_tensors.find(lt);
+                NPUW_ASSERT(iter_device_registered != device_bank.registered_tensors.end() &&
+                            "Tensor should be registered first!");
+                auto uid = iter_device_registered->second;
+                if (device_bank.storage[uid].tensor) {
```

Comment:
`device_bank.storage[uid]` would add a new item if it is not already present.

If this happening it is not safe to use `.at` without lock

Reply:
Removed all internal locks as discussed